enigma-principal/app/src/controller/km_controller.rs

```diff
@@ -55,39 +55,36 @@ impl KMController {
-        if receipt.logs.is_empty() {
-            // todo: change the web3error to a proper custom error
-            Err(Web3Error {
-                message: format!("A connection error occurred with the Smart Contract- workerParams did not return a log response" ),
-            })?
-        }
-        let log = receipt.logs[0].clone();
+        let log = receipt.logs.get(0).ok_or(Web3Error {
+            message: format!("A connection error occurred with the Smart Contract- workerParams did not return a log response" ),
+        })?.clone();
         let raw_log = RawLog { topics: log.topics, data: log.data.0 };
         let event = WorkersParameterizedEvent::new();
-        let result = match event.0.parse_log(raw_log) {
-            Ok(result) => result,
-            Err(err) => Err(Web3Error {
+        let result = match event.0.parse_log(raw_log).map_err(|err|
+            Web3Error {
                 message: format!("Unable to parse {} event: {:?}", WORKER_PARAMETERIZED_EVENT, err),
-            })?,
-        };
+            })?;
         debug!("Parsed the {} event: {:?}", WORKER_PARAMETERIZED_EVENT, result);
         Ok(result)
     }
 
     #[logfn(DEBUG)]
     fn verify_worker_params(&self) -> Result<(), ControllerError> {
-        for signed_epoch in self.epoch_verifier.get_all_confirmed().
-            map_err(ControllerError::VerifierError)?.iter() {
-            // if the epoch is confirmed by the Enigma Contract
-            if let Some(_) = &signed_epoch.confirmed_state {
-                // Get the km_block_number which indicates where to take the list of active workers from
-                let km_block_number = signed_epoch.get_km_block_num();
-                let (workers, stakes) = self.contract.get_active_workers(km_block_number).map_err(ControllerError::Other)?;
-                let worker_params = InputWorkerParams { km_block_number, workers, stakes };
-                set_or_verify_worker_params(self.eid, &worker_params, Some(signed_epoch.clone())).
-                    map_err(ControllerError::EnclaveError)?;
+        for signed_epoch in self.epoch_verifier
+            .get_all_confirmed()
+            .map_err(ControllerError::VerifierError)?
+            .iter()
+            {
+                // if the epoch is confirmed by the Enigma Contract
+                if signed_epoch.confirmed_state.is_some() {
+                    // Get the km_block_number which indicates where to take the list of active workers from
+                    let km_block_number = signed_epoch.get_km_block_num();
+                    let (workers, stakes) = self.contract.get_active_workers(km_block_number).map_err(ControllerError::Other)?;
+                    let worker_params = InputWorkerParams { km_block_number, workers, stakes };
+                    set_or_verify_worker_params(self.eid, &worker_params, Some(signed_epoch.clone())).
+                        map_err(ControllerError::EnclaveError)?;
+                }
```

Comment:
You didn't have to indent the code  inside the for loop itself, but it's not very important